lambda/lambda.mli

```diff
@@ -45,6 +45,24 @@ type is_safe =
   | Safe
   | Unsafe
 
+type field_read_semantics =
+  | Reads_agree
+  | Reads_vary
```

Comment:
What do these constructors mean?

Reply:
`Reads_agree` is a read from an immutable field, `Reads_vary` is a read from a mutable field.
I agree that this deserves a few documentation comments.